test/NuGet.Clients.Tests/NuGet.VisualStudio.Common.Test/NuGetExperimentationServiceTests.cs

```diff
@@ -3,121 +3,153 @@
 
 using System;
 using System.Collections.Generic;
-using System.Threading;
-using System.Threading.Tasks;
-using System.Windows.Media;
 using FluentAssertions;
 using Microsoft.VisualStudio.Experimentation;
+using Microsoft.VisualStudio.Sdk.TestFramework;
+using Moq;
 using NuGet.Common;
 using Test.Utility;
+using Test.Utility.VisualStudio;
 using Xunit;
 
 namespace NuGet.VisualStudio.Common.Test
 {
-    [Collection(nameof(TestJoinableTaskFactoryCollection))]
+    [Collection(MockedVS.Collection)]
     public class NuGetExperimentationServiceTests
     {
+        private readonly Mock<IOutputConsoleProvider> _outputConsoleProviderMock;
         private readonly Lazy<IOutputConsoleProvider> _outputConsoleProvider;
+        private readonly Mock<IOutputConsole> _outputConsoleMock;
 
-        public NuGetExperimentationServiceTests()
+        public NuGetExperimentationServiceTests(GlobalServiceProvider sp)
         {
-            _outputConsoleProvider = new Lazy<IOutputConsoleProvider>(() => new TestOutputConsoleProvider());
-        }
+            sp.Reset();
 
-        private IList<string> OutputMessages => ((TestOutputConsoleProvider)_outputConsoleProvider.Value).TestOutputConsole.Messages;
+            var mockOutputConsoleUtility = OutputConsoleUtility.GetMock();
+            _outputConsoleProviderMock = mockOutputConsoleUtility.mockIOutputConsoleProvider;
+            _outputConsoleProvider = new Lazy<IOutputConsoleProvider>(() => _outputConsoleProviderMock.Object);
+            _outputConsoleMock = mockOutputConsoleUtility.mockIOutputConsole;
+        }
 
         [Fact]
-        public void Constructor_WithNullWrapper_Throws()
+        public void Constructor_WithNullEnvironmentVariableReader_Throws()
```

Comment:
Do we care if a unit test is renamed?  We lose history in AzDevOps but that's not too big of a deal, just wondering.

Reply:
Interesting question - I've never heard anyone mention that concern. Since builds are only kept 30-60 days, I'd lean towards it not being a problem.